extension/parquet/parquet_extension.cpp

```diff
@@ -45,8 +45,14 @@
 namespace duckdb {
 
 struct ParquetReadBindData : public TableFunctionData {
+	// TODO: the file list mutex is currently required to do thread-safe access into the file_list in the bind_data
+	unique_ptr<mutex> file_list_lock;
+	unique_ptr<MultiFileList> file_list;
+	unique_ptr<MultiFileReader> multi_file_reader;
+	string first_file;
+	idx_t total_file_count;
+
 	shared_ptr<ParquetReader> initial_reader;
-	vector<string> files;
```

Comment:
Nightly test run here - https://github.com/samansmink/duckdb/actions/runs/8908217982 - PTAL at the threadsan